docs/enhancements/backup-restore-support.md

```diff
@@ -0,0 +1,151 @@
+I.e. Extract the current status of the BMH, store it in an annotation, create the BMH on the new hub:
+```
+# Save the status in json format to a file
+kubectl get bmh <name-of-bmh> -o jsonpath="{.status}" > status.json
+# Save the BMH and apply the status annotation to the saved BMH.
+kubectl -n metal3 annotate bmh <name-of-bmh> \
+  baremetalhost.metal3.io/status="$(cat status.json)" \
+  --dry-run=client -o yaml > bmh.yaml
+```
+This flow ensures that the BMH on the new hub has status (BMO applies the status using the annotation). Hence, as the BMH should be in state 'Provisioned', the BMH reconcile is avoided.
+I.e. prevent destruction of restored spoke clusters.
+
+##### Suggested solution for automating the flow (make it transparent to the user):
+
+For BMHs in the active hub: (In assisted-service -> `bmh_agent_controller`)
+* Once the BMH is in 'Provisioned' state:
+  * Extract BMH's status and add `baremetalhost.metal3.io/status` to the object metadata.
+  * Add also, add the `baremetalhost.metal3.io/paused` annotation:
+    * This is required since once the annotation is added, BMO updates the BMH's status and delete the annotation.
+    * The [paused](https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md#pausing-reconciliation) annotation, is similar to the `detached` annotation we're already using - but, it's also pausing the reconcile loop. Thus, keeping the status annotation intact.
+
+For BMHs in the passive hub: (In assisted-service -> `bmh_agent_controller`)
+* When the BMH is applied, the status will be missing - since reconcile is paused.
+  * To restore the original status:
+    * Remove the `baremetalhost.metal3.io/paused` annotation
+      * Status is restored, but `baremetalhost.metal3.io/status` annotation is removed.
+    * Hence, add the status and paused annotation back
+      * By detecting a BMH with `provisioned` state and a missing status annotation.  
+
+#### Solution for deleted Agent CRs
+
+1. Don't delete the Agent CR when an associated host doesn't exits on DB.
+   * Remove relevant [code](https://github.com/openshift/assisted-service/blob/5bbf696fc0cc5a220fd8a0d1e408ab0bcc889e3f/internal/controller/controllers/agent_controller.go#L143-L151) from AgentController.
+
+2. Restore the Host record in DB according to properties from the Agent CR.
+   * Seems we can infer more than the mandatory properties:
+     * host.ID <-- agent.Name
+     * host.Role <-- agent.Role
+     * host.InfraEnvID <-- agent.metadata.annotations['infraenvs.agent-install.openshift.io']
+     * host.Kind <-- 'Host' or 'AddToExistingClusterHost' if cluster is already installed
+     * host.InstallerArgs <-- agent.InstallerArgs
+     * host.RequestedHostname <-- agent.Hostname
+   * Some of the inventory data can be restored from annotations.
```

Comment:
Do we not have the entire inventory in agent.status? Are we expecting that to not be recreated on restore?

If not, should we consider taking BMHs example and create a similar status annotation?

Reply:
We shouldn't be writing status - that should only be done by the controllers.  We can consider a status annotation, but:
- We'll have status for uninstalled hosts anyway once they boot with an ISO for the new hub
- The status for installed hosts may be stale and they should use information collected from OpenShift on the spoke